backend/src/main/java/com/hackner/musicompass/service/ArtistSearchService.java

```diff
@@ -0,0 +1,45 @@
+package com.hackner.musicompass.service;
+import com.hackner.musicompass.discogsapi.model.DiscogsArtist;
+import com.hackner.musicompass.discogsapi.model.DiscogsArtistSearchResults;
+import com.hackner.musicompass.model.Artist;
+import com.hackner.musicompass.model.ArtistInfo;
+import com.hackner.musicompass.discogsapi.service.DiscogsArtistApiService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class ArtistSearchService {
+    private final DiscogsArtistApiService discogsArtistApiService;
+
+    @Autowired
+    public ArtistSearchService(DiscogsArtistApiService discogsArtistApiService) {
+        this.discogsArtistApiService = discogsArtistApiService;
+    }
+
+    public List<Artist> getArtistInfoListBySearchTerm(String searchTerm) {
+        DiscogsArtistSearchResults discogsArtistSearchResults = discogsArtistApiService.getDiscogsArtistListBySearchTerm(searchTerm);
+
+        List<DiscogsArtist> discogsArtistList = discogsArtistSearchResults.getResults();
+
+        if (discogsArtistList.size() > 5) {
+            discogsArtistList = discogsArtistList.subList(0, 5);
+        }
+
+        List<Artist> artistList = discogsArtistList.stream()
+                .map(discogsArtist -> {
+                    Artist artist = new Artist().builder()
+                            .artistName(discogsArtist.getArtistName())
+                            .artistInfo(new ArtistInfo().builder()
+                                    .artistImageUrl(discogsArtist.getArtistImageUrl())
+                                    .discogsArtistId(discogsArtist.getDiscogsArtistId())
+                                    .discogsArtistUrl(discogsArtist.getDiscogsArtistUrl()).build())
+                            .build();
+                    return artist;
+                })
+                .collect(Collectors.toList());
+
+        return artistList;
```

Comment:
```suggestion
        return discogsArtistList.stream()
                .map(discogsArtist -> new Artist().builder()
                            .artistName(discogsArtist.getArtistName())
                            .artistInfo(new ArtistInfo().builder()
                                    .artistImageUrl(discogsArtist.getArtistImageUrl())
                                    .discogsArtistId(discogsArtist.getDiscogsArtistId())
                                    .discogsArtistUrl(discogsArtist.getDiscogsArtistUrl()).build())
                            .build())
                .collect(Collectors.toList());
```

Reply:
you could also move 
```
new Artist().builder() .artistName(discogsArtist.getArtistName())
                            .artistInfo(new ArtistInfo().builder()
                                    .artistImageUrl(discogsArtist.getArtistImageUrl())
                                    .discogsArtistId(discogsArtist.getDiscogsArtistId())
                                    .discogsArtistUrl(discogsArtist.getDiscogsArtistUrl()).build())
                            .build()
```

to a map Artist method

Reply:
I will use the same method also in my second service class (ArtistService.java). Is it fine to create the method here and then import it in the second class? Should I rather create a class "ArtistHelper.java" just to create this method? What is elegantbest practice to use one method in two service classes?